Upstream-Status: Inappropriate [musl support]

mallinfo() is a glibc extension; stop trying to use it with musl.

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
diff --git a/base/process/process_metrics_posix.cc b/base/process/process_metrics_posix.cc
index 0eb5c1f97b0d..3404772f7994 100644
--- a/base/process/process_metrics_posix.cc
+++ b/base/process/process_metrics_posix.cc
@@ -94,7 +94,7 @@ size_t ProcessMetrics::GetMallocUsage() {
   malloc_statistics_t stats = {0};
   malloc_zone_statistics(nullptr, &stats);
   return stats.size_in_use;
-#elif defined(OS_LINUX) || defined(OS_ANDROID)
+#elif (defined(OS_LINUX) && defined(__GLIBC__)) || defined(OS_ANDROID)
   struct mallinfo minfo = mallinfo();
 #if defined(USE_TCMALLOC)
   return minfo.uordblks;
@@ -104,6 +104,9 @@ size_t ProcessMetrics::GetMallocUsage() {
 #elif defined(OS_FUCHSIA)
   // TODO(fuchsia): Not currently exposed. https://crbug.com/735087.
   return 0;
+#else
+  // musl.
+  return 0;
 #endif
 }
 
